src/CountryHandler/Slovakia.php

```diff
@@ -24,12 +26,28 @@ final class Slovakia extends CountryHandler
      */
     public const LENGTH = 10;
 
+    public const PATTERN = '([1-9]\\d[234789]\\d{7})|(\\d{2}[0156]\\d[0-3]\\d{4,5})';
+
+    public function hasValidRule(string $tin): bool
+    {
+        if (strlen($tin) === 10
+            && (int) $tin % 11 === 0) {
+            return true;
+        }
+
+        if (strlen($tin) === 10) {
+            return (int) substr($tin, 9, 1) === ((int) substr($tin, 0, 9) % 11) % 10;
+        }
+
+        return strlen($tin) === 9;
```

Comment:
```suggestion
        if (strlen($tin) === 10) {
          if ((int) $tin % 11 === 0) {
            return true;
          }

          return (int) substr($tin, 9, 1) === ((int) substr($tin, 0, 9) % 11) % 10;
        }

        return strlen($tin) === 9;
```

Reply:
(don't forget to run `phpcsfixer` after that change)

Reply:
Adapted your solution in my latest commit.

Reply:
I don't think so, or I misunderstand what you're doing.

There are still 2 calls to `if (strlen($tin) === 10)` and we definitely should avoid repeating ourselves.

Reply:
Commit was not pushed 👍

Reply:
I also removed the strlen check as we did that before in the matchLength.

Reply:
That was my next concern :)

Reply:
Should be pushed now :)